File: tournaments/divNumber/divNumber.py
def divNumber(k, l, r):
    ans = 0

    for i in range(l, r + 1):
        n = i
        divs = 1
        j = 2

        while n >= j * j:
            p = 0
            while n % j == 0:
                n /= j
                p += 1
            divs *= p + 1 
            if k < divs:
                break
            j += 1
        if 1 < n:
            divs *= 2
        if k == divs:
            ans += 1

    return ans

File: tournaments/divNumber/divNumber.cpp
int divNumber(int k, int l, int r) {

  int ans = 0;
  for (int i = l; i <= r; i++) {
    int n = i;
    int divs = 1;
    for (int j = 2; j * j <= n; j++) {
      int pow = 0;
      while (n % j == 0) {
        n /= j;
        pow++;
      }
      divs *= 1 + pow;
      if (divs > k) {
        break;
      }
    }
    if (1 < n) {
        divs *= 2;
    }
    if (divs == k) {
      ans++;
    }
  }

  return ans;
}
